controller/backup_volume_controller.go

```diff
@@ -276,15 +274,15 @@ func (bvc *BackupVolumeController) reconcile(backupVolumeName string) (err error
 	defer engineClientProxy.Close()
 
 	// Get a list of all the backups that are stored in the backup target
-	res, err := backupTargetClient.BackupNameList(backupTargetClient.URL, backupVolumeName, backupTargetClient.Credential)
+	res, err := backupTargetClient.BackupNameList(backupTargetClient.URL, remoteBackupVolumeName, backupTargetClient.Credential)
 	if err != nil {
 		log.WithError(err).Error("Failed to list backups from backup target")
 		return nil // Ignore error to prevent enqueue
 	}
 	backupStoreBackups := sets.NewString(res...)
 
 	// Get a list of all the backups that exist as custom resources in the cluster
-	clusterBackups, err := bvc.ds.ListBackupsWithBackupVolumeName(backupVolumeName)
+	clusterBackups, err := bvc.ds.ListBackupsWithBackupVolumeNameRO(backupTargetName, remoteBackupVolumeName)
```

Comment:
Any potential risk after changing it to a read-only list api?

Reply:
backups of `clusterBackups` will only be deleted if they are not on the remote backup target and will not be modified. There is no any potential risk on my mind now.